walker.go

```diff
@@ -199,29 +202,28 @@ func (s *StatInfo) Sys() interface{} {
 	return s.Stat
 }
 
-func patternPrefixes(patterns []string) []string {
-	pfxs := make([]string, 0, len(patterns))
-	for _, ptrn := range patterns {
-		idx := strings.IndexFunc(ptrn, func(ch rune) bool {
-			return ch == '*' || ch == '?' || ch == '[' || ch == '\\'
-		})
-		if idx == -1 {
-			idx = len(ptrn)
-		}
-		pfxs = append(pfxs, ptrn[:idx])
+func matchPrefix(pattern, name string) (bool, bool) {
```

Comment:
Should this return a "match" for `matchPrefix("/hello/*", "/hello/foo/world")`?

Reply:
this is never reached as `hello/foo` would match and skip the whole directory